python/ray/serve/_private/serve_request_response.py

```diff
@@ -0,0 +1,187 @@
+from abc import ABC, abstractmethod
+import grpc
+import logging
+import pickle
+
+from starlette.types import Receive, Scope, Send
+from typing import Any, List, Generator, Optional, Tuple
+
+from ray.actor import ActorHandle
+from ray.serve._private.constants import SERVE_LOGGER_NAME
+from ray.serve._private.utils import DEFAULT
+from ray.serve._private.common import gRPCRequest, StreamingHTTPRequest
+
+logger = logging.getLogger(SERVE_LOGGER_NAME)
+
+
+class ServeRequest(ABC):
```

Comment:
`ServeRequest` is very generic and doesn't help me understand what this is used for.

Generally when naming things, one trick is to describe the object in one sentence and it'll help you discover the natural name.

In this case you wrote down below: "to use in the common interface among proxies," which indicates to me that the "right" name is "ProxyRequest" :)

The subclass names flow naturally then: `HTTPProxyRequest`, `gRPCProxyRequest`

Reply:
I'm the worst in naming things XD Those are great suggestions, let me update them!